File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
    bool isvowel(char ch){
        ch = tolower(ch);
        return (ch=='a' || ch=='e' || ch =='i' || ch=='o' || ch=='u');
    }
public:
    string reverseVowels(string s) {
        int i = 0;
        int j  =s.size()-1;

        while(i<j){
            while(i<j && !isvowel(s[i])){
                i++;
            }
            while(i<j && !isvowel(s[j])){
                j--;
            }
            if(i<j){
                swap(s[i],s[j]);
                    i++;
                    j--;
                
            }
        }
        return s;
        
        
    }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    boolean isvowel(char ch){
        ch = Character.toLowerCase(ch);
        return (ch == 'a' || ch =='e' || ch=='i' || ch=='o' || ch=='u');
    }
    public String reverseVowels(String s) {
        char[] arr = s.toCharArray();
        int i = 0;
        int j = s.length()-1;
        while(i<j){
            while(i<j && !isvowel(arr[i])){
                i++;
            }
            while(i<j && !isvowel(arr[j])){
                j--;
            }
            if(i<j){
                char temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
                i++;
                j--;
            }
        }
        return new String(arr);
        
    }
}